be2-scala/src/main/scala/ch/epfl/pop/model/objects/Channel.scala

```diff
@@ -11,7 +11,7 @@ final case class Channel(channel: String) {
    */
   def decodeChannelLaoId: Option[Base64Data] = channel match {
     case _ if channel.startsWith(Channel.ROOT_CHANNEL_PREFIX) =>
-      Try(Base64Data(channel.substring(Channel.ROOT_CHANNEL_PREFIX.length).split(Channel.SEPARATOR)(0))) match {
+      Try(Base64Data(channel.substring(Channel.ROOT_CHANNEL_PREFIX.length).split(Channel.CHANNEL_SEPARATOR)(0))) match {
```

Comment:
Cannot highlight the right line (2 lines above), but is there a reason why you guys decided last semester to return a Base64Data optional here (instead of Hash optional) since we are returning a lao id?